lib/index.coffee

```diff
@@ -49,21 +49,11 @@ exports.create = (string, options, callback) ->
       return callback(error)
 
     file = Buffer.concat(stdout)
-    isFileBuffer = /^\%PDF/.test(file.slice(0, 4).toString())
-
-    if options.filename
-      callback(null, file.toString())
-
-    else if !isFileBuffer
-      filename = file.toString()
-      fs.readFile filename, (err, buffer) ->
-        return callback(err) if err
-        fs.unlink filename, (err) ->
-          callback(err, buffer)
-
-    else
-      callback(null, file)
 
+    try
+      file = JSON.parse(file)
+    catch err
```

Comment:
That's what I thought first.
But if you use a custom script that return only text but not JSON?
This functionality is just a shortcut if the result send back from PhantomJS can be parsed into an object or not.
For example, I'm using the default script, but also another custom script which send back a number, I don't want to have an error if JSON.parse(5) fail for example.